src/Tracing/Span.php

```diff
@@ -476,6 +483,60 @@ public function detachSpanRecorder()
         return $this;
     }
 
+    /**
+     * @return array<string, mixed>
+     */
+    public function getMetricsSummary(): array
+    {
+        return $this->metricsSummary;
+    }
+
+    /**
+     * @param string|int|float $value
+     * @param string[]         $tags
+     */
+    public function setMetricsSummary(
+        string $type,
+        string $key,
+        $value,
+        MetricsUnit $unit,
+        array $tags
+    ): void {
+        $mri = sprintf('%s:%s@%s', $type, $key, $unit);
+        $bucketKey = $mri . implode('', $tags);
+
+        if (\array_key_exists($bucketKey, $this->metricsSummary)) {
+            if ($type === SetType::TYPE) {
+                $value = 1.0;
+            } else {
+                $value = (float) $value;
+            }
+
+            $summary = $this->metricsSummary[$bucketKey];
+            $this->metricsSummary[$bucketKey] = [
+                'min' => min($summary['min'], $value),
+                'max' => max($summary['max'], $value),
+                'sum' => $summary['sum'] + $value,
+                'count' => $summary['count']++,
```

Comment:
Should be `$summary['count'] + 1`? The current code raises the value in the original summary to then override it with the old value.